### ISSUE TITLE
[FEAT] Add indicator for data written to disk

### ISSUE THREAD

**Issue by MeesJ** [AnalogJ/scrutiny#458]
**Is your feature request related to a problem? Please describe.**
In order to be able to easily monitor an SSD's warranty, the amount of terabytes written is very valuable. Currently, you can only see the raw LBAs written in Scrutiny, which requires manual calculating in order to extract the amount of data written to the storage device from it.

**Describe the solution you'd like**
In order to make it easier to check how many data has been written to an SSD, Scrutiny should display a mark that indicates the amount of GBs or TBs next to the marks that indicate the temperature, capacity and power-on time.

**Additional context**
![Concept of my idea](https://user-images.githubusercontent.com/77725857/225433451-fe48fc29-c82c-40e1-adca-4a86cc9e25ba.png)

**Comment by AnalogJ** [comment]
This would definitely be relevant to SSD's, though only if its in context (data written compared to warranty/disk EOL limit). I don't know if those limits are available anywhere.

**Comment by MeesJ** [comment]
These limits usually aren't included in the SMART data as far as I can tell, but users can definitely find them on the internet themselves. Perhaps you could allow users to set the TBW limit for their SDD themselves in Scrutiny?

**Comment by AnalogJ** [comment]
:grimacing: unfortunately there's not a nice way to do that, and we're left with a couple not-so-great options:

- drive configuration is done via the collector.yaml, which runs on the collector node(s), which sends limited data.
     - the collector only sends SMART data to the api, nothing else (at the moment). 
- adding this as a configuration option in the WebUI would probably work, however this would be the first option that is **only** configurable via the WebUI -- which I don't love either. 


I'm open to ideas though.

**Comment by MeesJ** [comment]
Perhaps the Wear Leveling Count SMART attribute could supply some useful information in this one? Otherwise, I'd think a TBW counter would still be useful for SSDs, even if the TBW warranty indication is unknown. Currently the only way to retrieve the amount of TBW via Scrutiny is by grabbing the LBA written attribute and manually calculating it to an amount of terabytes written.

**Comment by AnalogJ** [comment]
Yeah I guess in this case perfect is the enemy of good. Optionally displaying total bytes written for SSD's would still be valuable.

**Comment by BoKKeR** [comment]
In the case of servers where the data is written regularly it would be beneficial to be able to track the data written with a time graph. It would really help with NVME drives which have very low life and can wear out if there is excessive data written daily. 

![image](https://github.com/AnalogJ/scrutiny/assets/2664857/1fc3ed08-3f66-4246-91e5-a18ce1938694)

**Comment by ajaskiewiczpl** [comment]
Is this feaature still considered to be implemented? I just started using Scrutiny and was a little bit surprised it doesn't show TBW like any other SMART reporting software :)

**Comment by kafisatz** [comment]
I agree with @ajaskiewiczpl.
For reference proxmox seems to show a **wearout** value in the disks summary. 
Synology shows 'estimated endurance' (but it is somewhat cumbersome to find and only shown for the selected device)

**Comment by MeesJ** [comment]
> For reference proxmox seems to show a wearout value in the disks summary.

As a reference, this is how Proxmox lists the wearout value.
![](https://github.com/user-attachments/assets/e170a9bd-5522-4e65-8327-462132085622)

**Comment by cammurray** [comment]
Yeh the reason I need scrutiny is to monitor drive wearout for SSDs but it appears that's not available yet. Any other alternative solutions to monitor drive wearout that anyone's aware of?

**Comment by Turiok** [comment]
Hi,

In my scrutiny, the info is present but the name is unknown.
If the attribute is not known. Maybe use the one from smarctl.

And like every manufacturer, it has its limits. It should be possible to modify this in the collector.

<img width="1009" height="250" alt="Image" src="https://github.com/user-attachments/assets/5e71cce5-ebe0-41ac-b029-624d2024a550" />

**Comment by jwidess** [comment]
I've added a WIP version of this in my PR #847 however, I need to correct the problem of 512 vs 4096 block sizes. My PR was just a quick add for my NVMe drive. 
In the meantime, if anybody is looking to track drive SMART health data, including data written, I created a shell script to be used with Prometheus and node_exporter, specifically on OPNsense, but can easily be adopted to other platforms. https://github.com/jwidess/OPNsense-node-exporter-smartctl-collect

**Comment by kaysond** [comment]
This issue is being closed automatically as a part of an effort to streamline and prioritize development effort. If it is still relevant, please reply to this comment and we will convert it to a discussion.
